### PR TITLE
Small Tool and Workflow Refactoring and Fixes

### PR DESCRIPTION
**Pull request by jmchilton**
These changes don't really affect much but they cleanup huge changes to workflow and tool execution stuff from downstream CWL work [here](https://github.com/galaxyproject/galaxy/compare/dev...common-workflow-language:cwl).

### CODE REVIEW

**Comment by jmchilton** [comment]
Rebased a fix to 165f300 in there and added a bunch of new commits adding unit testing to the workflowprogress class.

File: lib/galaxy/managers/workflows.py [hunk 1]
```diff
@@ -544,19 +546,21 @@ def _workflow_to_dict_instance(self, trans, stored):
...
-            steps[step.id] = {'id': step.id,
-                              'type': step.type,
-                              'tool_id': step.tool_id,
-                              'tool_version': step.tool_version,
-                              'annotation': self.get_item_annotation_str( trans.sa_session, stored.user, step ),
-                              'tool_inputs': step.tool_inputs,
-                              'input_steps': {}}
+            step_type = step.type
+            step_dict = {'id': step.id,
+                         'type': step_type,
```
Comment (nsoranzo) [inline lib/galaxy/managers/workflows.py:554]: `step_type` seems to be used only here, can be reverted to `step.type` .

Comment (jmchilton) [inline lib/galaxy/managers/workflows.py:554]: The whole point of that commit is to clean up the diff for a larger feature addition downstream ([subworkflows](https://github.com/common-workflow-language/galaxy/commit/71cf09e8ff2bf640e2287ad0cd5426ae198f2dcc)) which does reuse the variable so I'd prefer to keep it this way, it isnt hurting anything.

Comment (nsoranzo) [inline lib/galaxy/managers/workflows.py:554]: Sure!

**Comment by nsoranzo** [comment]
:+1:

**Comment by nsoranzo** [comment]
I ran the api tests locally and I got the same 2 timeout errors of Jenkins, so they may be related to this PR.

**Comment by martenson** [comment]
@jmchilton Increase timeout? Moar `sync` ?

**Comment by jmchilton** [comment]
Looking over the logs, it looks like I legitimately broke something - I'll see if I can fix it without increasing timeouts.

**Comment by jmchilton** [comment]
Pushed 5c8a882 which fixes the failing test cases.

**Comment by nsoranzo** [comment]
Thanks @jmchilton!

**Comment by jmchilton** [comment]
Awesome, thanks for the comments and merge @martenson and @nsoranzo.

**Pull request merged**